fendermint/app/src/cmd/objects.rs

```diff
@@ -881,6 +887,19 @@ async fn os_get<F: QueryClient + Send + Sync>(
     Ok(return_data)
 }
 
+fn get_filename_with_extension(filename: &str, content_type: &str) -> Option<String> {
+    let path = Path::new(filename);
+
+    // Checks if filename already has extension
+    if let Some(_) = path.extension().and_then(|ext| ext.to_str()) {
+        return Some(filename.to_string());
+    }
+
+    get_mime_extensions_str(content_type)?
```

Comment:
If the filename does not have an extension, try to guess the extension. this can't be perfect, especially for `octet-stream` that can have multiple extensions for a given mime type